rdkit/Chem/Draw/IPythonConsoleIntegration.py

```diff
@@ -0,0 +1,424 @@
+        globals()['background_'+uid] = Dropdown(description='', options= bgcolors_3d, value=bgcolors_3d[1])
+        wgListBox.append(Box([Label(value='color'),globals()['colorScheme_'+uid]], layout=itemLayout))
+        wgListBox.append(Box([Label(value='background'),globals()['background_'+uid]], layout=itemLayout))
+        globals()['rdkit_wg_dict'].update({globals()['colorScheme_'+uid]._model_id:uid})
+        globals()['rdkit_wg_dict'].update({globals()['background_'+uid]._model_id:uid})
+        globals()['colorScheme_'+uid].observe(handle_change, names='value')
+        globals()['background_'+uid].observe(handle_change, names='value')
+    
+    
+    # labels
+    if labelPanel is True:
+        globals()['confLabel_'+uid] = Checkbox(description='confLabel', value=False)
+        globals()['atomLabel_'+uid] = Checkbox(description='atomLabel', value=False)
+        
+        cbLabel=Box([Label(value=''),
+                     HBox([globals()['confLabel_'+uid], 
+                           globals()['atomLabel_'+uid]])
+                    ], layout=itemLayout)
+        
+        wgListBox.append(cbLabel)
+        
+        globals()['rdkit_wg_dict'].update({globals()['confLabel_'+uid]._model_id:uid})
+        globals()['rdkit_wg_dict'].update({globals()['atomLabel_'+uid]._model_id:uid})
+        globals()['confLabel_'+uid].observe(handle_change, names='value')
+        globals()['atomLabel_'+uid].observe(handle_change, names='value')
+    
+    
+    # buttons
+    globals()['start_'+uid] = Button(description="Start!", button_style='success')
+    globals()['zoomTo_'+uid] = Button(description="zoomTo", button_style='success')
+    buttons=Box([Label(value=''),HBox([globals()['zoomTo_'+uid], globals()['start_'+uid]])],layout=itemLayout)
+    wgListBox.append(buttons)
+    
+    globals()['rdkit_wg_dict'].update({globals()['start_'+uid]._model_id:uid})
+    globals()['rdkit_wg_dict'].update({globals()['zoomTo_'+uid]._model_id:uid})
+    globals()['start_'+uid].on_click(handle_start_button)
+    globals()['zoomTo_'+uid].on_click(handle_zoomTo_button)
+    
+    
+    # left panel (container holding table for viewer)
+    size = (435, 485)
+    viewerLayout=Layout(width=str(size[0]+5)+'px', height=str(size[1]+5)+'px', border='solid')
+    wgLeftWidget = HTML('''<table><tr><td id="%s"></td></tr></table>'''%uid, layout=viewerLayout)
+    
+    # right panel
+    wgRightBox=VBox(wgListBox, layout=Layout(border='solid', width='45%'))
+    
+    # Combining left and right panel
+    all_wg = HBox([wgLeftWidget, wgRightBox])
+    
+    # displaying everything
+    display(all_wg)
+    
+    
+    # inserting 3DMol.js viewer in existing container (table)
+    globals()['view_'+uid] = py3Dmol.view(width=size[0],height=size[1])
+    globals()['view_'+uid].setBackgroundColor('0x000000')
+    globals()['view_'+uid].zoomTo()
+    
+    display(globals()['view_'+uid].insert(uid))
+    
```

Comment:
Suggestion for combining everything that describes the molecule view state:

```python
class MolViewState(object):
  def __init__(self, molecules):
    """ Molecules is dictionary of molecules """
    self.uid = str(time.time()).replace('.','')
    self.moldict = molecules

    # These should have reasonable initial values
    self.rdkit_mol_select = set()
    self.rdkit_conf_select = set()
    self.allConfIDs = []

  def selectMolecules(self, selectAllMols, selectMultiMols, selectMol):
    """ Select either all moleculs or add selectMol or show only selectMol """
    if selectAllMols:
       self.rdkit_mol_select = set(self.moldict)
    elif selectMultiMols:
      self.rdkit_mol_select.add(selectMol)
    else:
      self.rdkit_mol_select = {selectMol}

   def selectConformations(self, selectAllConfs, selectMultiConfs, selectConf):
    """ For all selected molecules, select either all conformations or add selectConf or show only selectConf """
    for mol in self.selectedMolecules:
      nConformers = mol.GetNumConformers()
      if nConformers > 1:
        if selectAllConfs:
          self.rdkit_conf_select = set(range(nConformers))
        elif selectMultiConfs:
          self.rdkit_conf_select.add(selectConf)
        else:
          self.rdkit_conf_select = {selectConf}
      elif mol.GetNumConformers() == 1:
        self.rdkit_conf_select = {0}

  @property
  def selectedMolNames(self):
    """ Return the names of all selected molecules """
    return self.rdkit_mol_select

  @property
  def selectedConfIds(self):
    """ Return the names of all selected molecules """
    return self.rdkit_conf_select

  @property
  def selectedMolecules(self):
    """ Return the selected molecules """
    return [self.moldict[name] for name in self.selectedMolNames]

  @property
  def selectedModels(self):
    """ Iterator over all selected models (molecules/conformations) """
    for mol in self.selectedMolecules:
      for confId in self.selectedConfIds:
        yield Chem.MolToMolBlock(mol, confId=confId)

  @property
  def allConfIds(self):
    """ Return the number of conformations - use the first selected molecule to determine """
    nconfIds = self.selectedMolecules[0].GetNumConformers()
    return list(range(nconfIDs))
```
and initialize like this:
```
molViewState = MolViewState(ProcessMolContainingObj(Mol))
globals()['mol_views'][molViewState.uid] = molViewState
```
The class is now independent of the viewer and you can easily write tests for it. It will also be easier to modify it should it be necessary to support different number of conformers. I would also add the property calculation to this class and actually move it into a separate module. As the ipython functionality becomes more and more important, consider having a separate directory for it ( @greglandrum please comment on this).


Using this class, it would lead to the following change in `update3D`:
```
uid = globals()['rdkit_wg_dict'][model_id]
molViewState  = globals()['mol_views'][uid]

molViewState.selectMolecules(globals()['selectAllMols_'+uid].value, 
                             globals()['selectMultiMols_'+uid].value,
                             globals()['molTupleId_'+uid].value)
globals()['selected_mols_view_'+uid].value = ', '.join(molViewState.selectedMolNames)

molViewState.selectConformations(globals()['selectAllConfs_'+uid].value, 
                                 globals()['selectMultiConfs_'+uid].value,
                                 globals()['confId_'+uid].value)
globals()['confId_'+uid].options = molViewState.allConfIds()
globals()['selected_confs_view_'+uid].value = ', '.join([str(x) for x in confIdsList])

# Add models (molecules/conformations) to viewer
for model in molViewState.selectedModels:
  globals()['view_'+uid].addModel(model, 'sdf')
```
This should give you an idea how to continue with this. I would also combine everything that has to do with controls in one class and the 3D view of the molecule into one class as well.